pol-core/plib/mul/map.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include <stdexcept>
+
+namespace Pol::Plib::MUL
+{
+struct Map
+{
+  static constexpr size_t blockWidth = 8;
+  static constexpr size_t blockHeight = 8;
+  static constexpr size_t cellSize = 3;
+
+  // DWORD + 64 cells (3 bytes) per block
+  static constexpr size_t blockSize = 4 + cellSize * blockWidth * blockHeight;
+
+  static constexpr size_t expected_blocks( int width, int height )
+  {
+    return ( width / blockWidth ) * ( height / blockHeight );
+  }
+
+  static constexpr bool valid_size( size_t filesize, int width, int height )
+  {
+    if ( width % blockWidth != 0 )
+      return false;
+
+    if ( height % blockHeight != 0 )
+      return false;
+
+    size_t nblocks = filesize / Map::blockSize;
+    return expected_blocks( width, height ) == nblocks;
+  }
+};
+static_assert( Map::blockSize == 196, "Size mismatch" );
+static_assert( Map::valid_size( 77070336, 6144, 4096 ) );                 // Legacy map
```

Comment:
Maybe move these checks to the unittest and not as static_assert? blocksize check looks like something a platform could change due to type sizes so ok to test it as static_assert. But something like valid_size() is more a unittest.

Reply:
I decided to leave those in, though I added other tests. This is just a small sanity check that can be checked at compile time.